internal/cmd/commands/authmethodscmd/ldap_funcs.go

```diff
@@ -0,0 +1,440 @@
+				Name:   upnDomainFlagName,
+				Target: &c.flagUpnDomain,
+				Usage:  "The userPrincipalDomain used to construct the UPN string for the authenticating user (optional).",
+			})
+		case userDnFlagName:
+			f.StringVar(&base.StringVar{
+				Name:   userDnFlagName,
+				Target: &c.flagUserDn,
+				Usage:  "The base DN under which to perform user search (optional).",
+			})
+		case userAttrFlagName:
+			f.StringVar(&base.StringVar{
+				Name:   userAttrFlagName,
+				Target: &c.flagUserAttr,
+				Usage:  "The attribute on user entry matching the username passed when authenticating (optional).",
+			})
+		case userFilterFlagName:
+			f.StringVar(&base.StringVar{
+				Name:   userFilterFlagName,
+				Target: &c.flagUserFilter,
+				Usage:  "A go template used to construct a LDAP user search filter (optional).",
+			})
+		case enableGroupsFlagName:
+			f.BoolVar(&base.BoolVar{
+				Name:   enableGroupsFlagName,
+				Target: &c.flagEnableGroups,
+				Usage:  "Find the authenticated user's groups during authentication (optional).",
+			})
+		case groupDnFlagName:
+			f.StringVar(&base.StringVar{
+				Name:   groupDnFlagName,
+				Target: &c.flagGroupDn,
+				Usage:  "The base DN under which to perform group search.",
+			})
+		case groupAttrFlagName:
+			f.StringVar(&base.StringVar{
+				Name:   groupAttrFlagName,
+				Target: &c.flagGroupAttr,
+				Usage:  "The attribute that enumerates a user's group membership from entries returned by a group search (optional).",
+			})
+		case groupFilterFlagName:
+			f.StringVar(&base.StringVar{
+				Name:   groupFilterFlagName,
+				Target: &c.flagGroupFilter,
+				Usage:  "A go template used to construct a LDAP group search filter (optional).",
+			})
+		case certificatesFlagName:
+			f.StringSliceVar(&base.StringSliceVar{
+				Name:   certificatesFlagName,
+				Target: &c.flagCertificates,
+				Usage:  "PEM-encoded X.509 CA certificate that can be used as trust anchors when connecting to an LDAP server (optional). May be specified multiple times.",
+			})
+		case clientCertificateFlagName:
+			f.StringVar(&base.StringVar{
+				Name:   clientCertificateFlagName,
+				Target: &c.flagClientCertificate,
+				Usage:  "A client certificate in ASN.1 DER form encoded as PEM (optional).",
+			})
+		case clientCertificateKeyFlagName:
+			f.StringVar(&base.StringVar{
+				Name:   clientCertificateKeyFlagName,
+				Target: &c.flagClientCertificateKey,
+				Usage:  "A client certificate key in PKCS #8, ASN.1 DER form encoded as PEM (optional).",
+			})
+		case bindDnFlagName:
+			f.StringVar(&base.StringVar{
+				Name:   bindDnFlagName,
+				Target: &c.flagBindDn,
+				Usage:  "The distinguished name of entry to bind when performing user and group searches (optional).",
+			})
+		case bindPasswordFlagName:
+			f.StringVar(&base.StringVar{
+				Name:   bindPasswordFlagName,
+				Target: &c.flagBindPassword,
+				Usage:  "The password to use along with bind-dn performing user and group searches (optional).",
```

Comment:
Is this something we should only be reading from `file://` or `env://`?

Reply:
yes and the client cert key as well. ty!